vowpalwabbit/core/include/vw/core/object_pool.h

```diff
@@ -92,6 +93,7 @@ class no_lock_object_pool
     return size;
   }
 
+  VW_DEPRECATED("Pools will no longer be able to check if an object is from the pool in VW 10.")
```

Comment:
why is this longer not needed? what changed

Reply:
The removal of finish_example and VW's required managed pool makes this unnecessary. In fact, removing this makes the pool more flexible since you can add externally created object into the pool